app/models/setting.rb

```diff
@@ -229,6 +229,23 @@ def self.model_name
     ActiveModel::Name.new(Setting)
   end
 
+  # Define which settings are considered sensitive
+  def sensitive_setting?
+    %w[http_proxy oauth_consumer_secret smtp_password].include?(name)
+  end
```

Comment:
Instead of hardcoding a list sensitive settings, would it be possible to extend the setting dsl to allow marking an arbitrary setting as sensitive in the place where the setting is defined?

Reply:
> Instead of hardcoding a list sensitive settings, would it be possible to extend the setting dsl to allow marking an arbitrary setting as sensitive in the place where the setting is defined?

We already have `encrypted?` which is our current way of dealing with this. It's delegated to the definition so that should take care of it.

Reply:
Thanks Ewoud, I agree.